packages/flutter/lib/src/material/button.dart

```diff
@@ -231,14 +239,20 @@ class RawMaterialButton extends StatefulWidget {
 }
 
 class _RawMaterialButtonState extends State<RawMaterialButton> {
-  bool _highlight = false;
-  bool _focused = false;
-  bool _hovering = false;
+  final Set<MaterialState> _states = <MaterialState>{};
+
+  void _updateState(bool value, MaterialState state) {
+    if (value) {
```

Comment:
nit: ordering of this function

updateState(state, value) reads more like "update state to value"

Reply:
Good call, `updateState(state, value)` is much better!